katdal/concatdata.py

```diff
@@ -312,24 +310,6 @@ def __init__(self, caches, keep=None):
             actual.update(cache.items())
             virtual.update(cache.virtual)
             self.props.update(cache.props)
-        # Pad out actual sensors on each cache (replace with default sensor values where missing)
-        for name, sensor_data in actual.items():
-            for cache in caches:
-                if name not in cache:
-                    # The original "raw" sensor name can differ from the cache
-                    # name due to aliasing, and concatenation cares about it.
-                    # Fully extracted ndarrays don't have .name, though...
-                    raw_name = getattr(sensor_data, 'name', name)
-                    dtype = sensor_data.dtype
-                    if dtype is None:
-                        # Instantiate base class as placeholder if type unknown
-                        cache[name] = SensorData(raw_name, dtype)
-                    else:
-                        cache[name] = dummy_sensor_data(raw_name, dtype=dtype)
-        # Pad out virtual sensors with default functions (nans)
```

Comment:
This virtual sensor padding seems to be gone now (probably for the best). It looks like the last virtual sensor lambda will win the day (if they differ between the underlying datasets).